docs/features/plugin/tools/getting-started.md

```diff
@@ -0,0 +1,270 @@
+---
+sidebar_position: 5
+title: "🔧📖  Tool Development Guide"
+---
+# 🔧📖 Tool Development Guide
+This documentation provides a complete guide for developers to create tools compatible with Open WebUI. By following these guidelines, you can ensure that your tools are efficient, user-friendly, and integrate well with the platform.
+
+> **Tip:**  A great way to get more familiar with tools is by exploring [community tools](https://openwebui.com/tools/).
+---
+
+## Introduction
+Open WebUI allows developers to extend its functionality by building custom tools using **pure python**. To maintain consistency and usability across the platform, it is essential to follow the development standards outlined in this guide.
+
+---
+
+## Class Structure
+All tools must be encapsulated in a class named `Tools`. This ensures compatibility with Open WebUI’s architecture.
+
+```python 
+class Tools:
+    def __init__(self):
+        pass
+```
+### file_handler:
```

Comment:
Add a heading above this to mention these two are reserved `instance attributes`